aiogram_dialog/widgets/kbd/datecalendar.py

```diff
@@ -0,0 +1,151 @@
+from datetime import date, datetime
+from time import mktime
+from calendar import monthcalendar
+from typing import List, Callable, Union, Awaitable
+
+from aiogram.types import InlineKeyboardButton, CallbackQuery
+
+from aiogram_dialog.dialog import Dialog
+from aiogram_dialog.manager.manager import DialogManager
+from aiogram_dialog.manager.intent import ChatEvent
+from aiogram_dialog.widgets.widget_event import WidgetEventProcessor, ensure_event_processor
+from .base import Keyboard
+
+OnDateSelected = Callable[[ChatEvent, "MonthCalendar", DialogManager, date], Awaitable]
+
+YEARS_LEVEL = "Y"
+MONTHS_LEVEL = "M"
+
+class MonthCalendar(Keyboard):
+    def __init__(self,
+                 id: str,
+                 on_click: Union[OnDateSelected, WidgetEventProcessor, None] = None,
+                 when: Union[str, Callable] = None):
+        super().__init__(id, when)
+        self.on_click = ensure_event_processor(on_click)
+
+    async def render_keyboard(self, 
+                              data,
+                              manager: DialogManager) -> List[List[InlineKeyboardButton]]:
+        offset = self.get_offset(manager)
+        if manager.context.data(f"{self.widget_id}:LEVEL", None, internal=True) == YEARS_LEVEL:
+            rendered_kbd = self.years_kbd(offset)
+        elif manager.context.data(f"{self.widget_id}:LEVEL", None, internal=True) == MONTHS_LEVEL:
+            rendered_kbd = self.months_kbd(offset) 
+        else:
+            rendered_kbd = self.days_kbd(offset)
+
+        return rendered_kbd
+
+    async def process_callback(self, 
+                               c: CallbackQuery, 
+                               dialog: Dialog,
+                               manager: DialogManager) -> bool:
+        prefix = f"{self.widget_id}:"
+        offset = self.get_offset(manager)
+        if not c.data.startswith(prefix):
+            return False
+        data = c.data[len(prefix):]
+        if data == "+":
+            self.set_offset(datetime(offset.year + (offset.month // 12), ((offset.month % 12) + 1), 1), manager)
+        elif data == "-":
+            if offset.month == 1:
+                self.set_offset(datetime(offset.year - 1, 12, 1), manager)
+            else:
+                self.set_offset(datetime(offset.year, (offset.month - 1), 1), manager)
+        elif data == MONTHS_LEVEL:
+            manager.context.set_data(f"{self.widget_id}:LEVEL", data, internal=True)
+        elif data == YEARS_LEVEL:
+            manager.context.set_data(f"{self.widget_id}:LEVEL", data, internal=True)
+        elif "month" in data:
+            data = int(c.data[len(prefix) + 6:])
+            manager.context.set_data(f"{self.widget_id}:LEVEL", False, internal=True)
```

Comment:
move `f"{self.widget_id}:LEVEL"` to attribute if current instance. It is magic literal and used more than once